x-pack/plugins/endpoint/public/applications/endpoint/view/alerts/details/overview/take_action_dropdown.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useMemo, useState } from 'react';
+import { EuiPopover, EuiFormRow, EuiButton, EuiLink, EuiIcon } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import styled from 'styled-components';
+
+export const TakeActionDropdown = styled(
+  memo(({ className }: { className?: string }) => {
+    const [isDropdownOpen, setIsDropdownOpen] = useState(false);
```

Comment:
@oatkiller @paul-tavares @dplumlee Wanted to get your thoughts on **local component state**. Here Davis is using local component state to keep track of whether the dropdown is open or closed. In the past, we have leaned towards keeping all application state, including local component state, in the redux store. It's a bit different now because we have been using EUI components heavily and they use local component state internally. So currently we don't store all application state in the redux store.

My main question for you is: For reusable components that **we** create, do we want to we want to store the state in the redux store or use local state?

If we wanted to store reusable component state in the redux store, we could create new versions of helpers like [tacotruck activatable](https://github.com/elastic/endgame-tacotruck/blob/dev/src/lib/redux-helpers/activatable_components.js).

Reply:
question. can the popover et al work without any external state?

Reply:
👍  I agree that for re-usable components, local state (or props) should be used to drive the component's functionality and for them to not have state stored in Redux (especially if there is a chance of these reusable components to be present on a view multiple times). 

💭  That being said - this is a fuzzy boundary (IMO). Normally (I think) redux/store is helpful when wanting to share state across the entire application, but the way we are using it is more as a global place to store state that applies only to the given view. One could also argue that we could use local state in that case as well (_especially around Views, which act more like Container Components_). I can tell you that as I work more with the Ingest code base, which is all built around local state and the use of hooks, the dev. mental model is allot simpler.

Reply:
I think this is fine. if some other component needs to know about the state then we should reconsider at that point. 
In this case, it's not much different than having a native `<select />` maintain its own open/closed state.